src/main/java/org/fcrepo/auth/webac/WebACRolesProvider.java

```diff
@@ -193,6 +198,64 @@ public void deleteRoles(final Node node) throws RepositoryException {
     };
 
     /**
+     *  This maps a Collection of acl:agentClass values to a List of agents.
+     *  Any out-of-domain URIs are silently ignored.
+     */
+    private List<String> dereferenceAgentClasses(final Collection<String> agentClasses) {
+        final Session internalSession = sessionFactory.getInternalSession();
+        final IdentifierConverter<Resource, FedoraResource> translator =
+                new DefaultIdentifierTranslator(internalSession);
+
+        final List<String> members = new ArrayList<>();
+        agentClasses.stream()
+                    .distinct()
+                    .forEach(x -> {
+                        if (x.startsWith(FEDORA_INTERNAL_PREFIX)) {
+                            final FedoraResource resource = nodeService.find(
+                                internalSession, x.substring(FEDORA_INTERNAL_PREFIX.length()));
+                            members.addAll(getFoafMembers(translator, resource));
+                        } else if (x.startsWith(FOAF_NAMESPACE_VALUE)) {
```

Comment:
This logic comes from the examples in the "spec". The W3C document provides two types of examples of `acl:agentClass`:

* the first class of examples includes references to an `rdfs:Class`, e.g. `<> acl:agentClass foaf:Agent .` (These do not require further dereferencing.)
* the second class of examples include a URI that can be dereferenced. These can be divided into two groups -- those URIs that are local and those that are external. We decided to exclude external references for now, so the logic only follows local URIs.

This does raise a question about what constitutes a valid target for an `acl:agentClass` resource. We could choose to also support `ore:aggregates`, but for now `foaf:member` seems sufficient, and so that's what has been implemented.